packages/documentation/src/stories/components/switch/switch.stories.tsx

```diff
@@ -0,0 +1,242 @@
+        'both',
+        'before',
+        'after'
+      ],
+      table: {
+        category: 'General'
+      }
+    },
+    labelBefore: {
+      name: 'Label (before)',
+      description: 'Describes the content/topic of the component.',
+      if: {
+        arg: 'labelPosition',
+        neq: 'after'
+      },
+      control: {
+        type: 'text'
+      },
+      table: {
+        category: 'General'
+      }
+    },
+    labelAfter: {
+      name: 'Label (after)',
+      description: 'Describes the content/topic of the component.',
+      if: {
+        arg: 'labelPosition',
+        neq: 'before'
+      },
+      control: {
+        type: 'text'
+      },
+      table: {
+        category: 'General'
+      }
+    },
+    hiddenLabel: {
+      name: 'Hidden Label',
+      description: '<p>Render the component with or without a visible label.</p><div className="alert alert-info alert-sm">There are accessibility issues with hidden labels.<br/>Please read our <a href="/?path=/story/foundations-accessibility--page#labels">labels accessibility guide</a>.</div>',
+      control: {
+        type: 'boolean'
+      },
+      table: {
+        category: 'General'
+      }
+    },
+    checked: {
+      name: 'Checked',
+      description: 'Defienes the checked state of the component.',
+      control: {
+        type: 'boolean'
+      },
+      table: {
+        category: 'States'
+      }
+    },
+    disabled: {
+      name: 'Disabled',
+      description: '<p>When set to `true`, disables the component\'s functionality and places it in a disabled state.</p><div className="alert alert-info alert-sm">There are accessibility issues with the disabled state.<br/>Please read our <a href="/?path=/docs/foundations-accessibility--page#disabled-state">disabled state accessibility guide</a>.</div>',
+      control: {
+        type: 'boolean'
+      },
+      table: {
+        category: 'States'
+      }
+    },
+    validation: {
+      name: 'Validation',
+      description: 'Controls the validation state appearance of the component.',
+      control: {
+        type: 'radio',
+        labels: {
+          'null': 'Default',
+          'is-valid': 'Valid',
+          'is-invalid': 'Invalid'
+        }
+      },
+      options: [
+        'null',
+        'is-valid',
+        'is-invalid'
+      ],
+      table: {
+        category: 'States'
+      }
+    },
+    validFeedback: {
+      name: 'Valid Feedback',
+      description: 'Text to show when the component is in an valid state.',
+      control: {
+        type: 'text'
+      },
+      table: {
+        category: 'States'
+      }
+    },
+    invalidFeedback: {
+      name: 'Invalid Feedback',
+      description: 'Text to show when the component is in an invalid state.',
+      control: {
+        type: 'text'
+      },
+      table: {
+        category: 'States'
+      }
+    }
+  }
+} as Meta;
+
+function toggle (args: Args, updateArgs: Function) {
+  updateArgs({ checked: !args.checked });
```

Comment:
This way of implementing the switch induces a reloading of the component on the documentation page, the animation is then incorrect.

Reply:
I see no way to fix that.
The only way we could get around that, is to remove the whole `checked` arg.

Reply:
Why not use `args.checked` to set the value of the checkbox `defaultChecked` property and not the `checked` property:
![image](https://user-images.githubusercontent.com/33580481/207569145-8457340d-72fa-4ca1-856c-253917e96795.png)

This would then behave exactly like with the HTML `checked` attribute: `defaultChecked` defines whether the checkbox is initially checked or not, then each time the user clicks the checkbox it natively toggles between a checked and an unchecked state without specifying an `onChange` handler.

Reply:
Your suggested solution, is a good approche, but I wanted to update the `checked` control in the `args-table` as well, whenever the user clicks on the switch in the preview panel.
To reach this behavior, I use your `defaultChecked` attribute approche (thanks for that 👍🏼) but keep the `onChange` event, which updates the control in the `args-table`.
I only had to add a 250ms timeout, to wait until the animation of the switch has finished.
Hope this is acceptable!?